File: framework/src/play-filters-helpers/src/main/java/play/filters/components/HttpFiltersComponents.java
/*
 * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
 */

package play.filters.components;

import play.components.HttpComponents;
import play.mvc.EssentialFilter;

import java.util.Arrays;
import java.util.List;

/**
 * A compile time default filters components.
 *
 * <p>Usage:</p>
 *
 * <pre>
 * public class MyComponents extends BuiltInComponentsFromContext
 *                           implements play.filters.components.HttpFiltersComponents {
 *
 *    public MyComponents(ApplicationLoader.Context context) {
 *        super(context);
 *    }
 *
 *    // required methods implementation
 *
 * }
 * </pre>
 *
 * @see NoHttpFiltersComponents
 */
public interface HttpFiltersComponents extends
        AllowedHostsComponents,
        CORSComponents,
        CSPComponents,
        CSRFComponents,
        GzipFilterComponents,
        RedirectHttpsComponents,
        SecurityHeadersComponents,
        HttpComponents {

    @Override
    default List<EssentialFilter> httpFilters() {
        return Arrays.asList(
            csrfFilter().asJava(),
            securityHeadersFilter().asJava(),
            allowedHostsFilter().asJava()
        );
    }
}

File: framework/src/play-filters-helpers/src/main/scala/play/filters/HttpFiltersComponents.scala
/*
 * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
 */

package play.filters

import play.api.mvc.EssentialFilter
import play.filters.csp.CSPComponents
import play.filters.csrf.CSRFComponents
import play.filters.headers.SecurityHeadersComponents
import play.filters.hosts.AllowedHostsComponents

/**
 * A compile time default filters components.
 *
 * {{{
 * class MyComponents(context: ApplicationLoader.Context)
 *   extends BuiltInComponentsFromContext(context)
 *   with play.filters.HttpFiltersComponents {
 *
 * }
 * }}}
 */
trait HttpFiltersComponents
  extends CSRFComponents
  with SecurityHeadersComponents
  with AllowedHostsComponents {

  def httpFilters: Seq[EssentialFilter] = Seq(csrfFilter, securityHeadersFilter, allowedHostsFilter)
}
